runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -4836,7 +4847,16 @@ genCAS(TR::Node *node, TR::CodeGenerator *cg, TR_ARM64ScratchRegisterManager *sr
       op = casWithoutSync ? (is64bit ? TR::InstOpCode::casx : TR::InstOpCode::casw) : (is64bit ? TR::InstOpCode::casalx : TR::InstOpCode::casalw);
       generateTrg1MemSrc1Instruction(cg, op, node, resultReg, TR::MemoryReference::createWithDisplacement(cg, addrReg, 0), newVReg);
       generateCompareInstruction(cg, node, resultReg, oldVReg, is64bit);
-      generateCSetInstruction(cg, node, resultReg, TR::CC_EQ);
+
+      if (!isExchange)
+         {
+         generateCSetInstruction(cg, node, resultReg, TR::CC_EQ);
+         }
+      else if (isReference && comp->target().is64Bit() && comp->useCompressedPointers())
```

Comment:
No need for `is64Bit()` here.  AArch64 always runs in the 64-bit mode.

Reply:
Please add description of the `isExchange` case in the comment above (lines 4840-4845).
I would appreciate if you also rewrite `oldReg` at line 4843 to `oldVReg`.

Reply:
I can change that.